Automattic/jetpack
# Carousel: show info icon even if Exif metadata option is turned off

#### Changes proposed in this Pull Request:

* If the `Show photo Exif metadata in carousel (when available)` option for the Carousel is turned off, then users will not be able to see the `(i)` info icon & important image information like the description as mentioned in: #20470
* In this PR, **the `jp-carousel-icon-info` is always shown**.
  * If users don't want to see the info icon, they could hide with CSS.
  * Exif metadata will not be displayed with this change if this option is off, but users will be able to see the description, title, and full size link.
* This also addresses the issue of not having having access to the `View full size` link if the Exif metadata option is turned off as mentioned in: #20291

Fixes #20291, #20470

#### Jetpack product discussion

We can discuss here, trying to mainly address the two issues above.

An aside, it might be interesting to try letting the info/comment toggles be sticky so you don't always need to click them after advancing an image, or maybe have a filter/option to expand by default. Which might somewhat address wanting to say show the title & description without needing to click the info icon each time (even if you have to scroll down a bit). 

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* After pulling down these changes on a JP connected site, in `/wp-admin/admin.php?page=jetpack#/writing` make sure that `Display images in a full-screen carousel gallery` is turned on, but that the `Show photo Exif metadata in carousel (when available)` is turned off.
* Create/edit a gallery, and make sure you have an image added that has metadata ([examples](https://github.com/ianare/exif-samples/tree/master/jpg))
* When the Carousel is opened, you should be able to see the `(i)` info icon, even though the Exif metadata option is off.
* Clicking on the info icon, you should be able to see the image title, caption, description, and "View full size" link. You should not be able to see the additional Exif metadata at this point.
* Turn back on the `Show photo Exif metadata in carousel (when available)` option, refresh, and make sure you can then see the metadata displayed.

#### Screenshots

**Before:**

![1](https://user-images.githubusercontent.com/15803018/129975698-00cc4895-b8d8-427a-8a74-bab3e19ddfb5.png)

---

**After:**

![2](https://user-images.githubusercontent.com/15803018/129975704-f764f838-292f-464f-bcf5-e5f7c786ec32.png)

---

**After (Exif metadata on):**
![3](https://user-images.githubusercontent.com/15803018/129975708-ba61f5ed-be43-4855-9e54-090aa8fa411b.png)
